### PR TITLE
update rect

### PR DESCRIPTION
**Pull request by Laughing-q**
@AyushExel @glenn-jocher 
now the `rect` support rect training if mode="train", and it support rect evaluation if mode="val".

- Train mode(training and evaluation) :
training: square images if rect=False(default), rect images if set rect=True.
valing: rect images

- Val mode(standalone evaluation):
valing: rect images by default, square images if model is not pt file or user manually set rect=False.

related issue: #792, #796

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced configuration flexibility and evaluation process in Ultralytics' pipeline.

### 📊 Key Changes
- Added `val` configuration to control validation/testing during training at a different documentation section.
- Expanded the functionality of the `rect` key to support rectangular evaluation during validation/testing, not just in training.
- Updated default behavior in `model.py` to enable rectangular batches by default during validation.
- Ensured that when using CPU inference or when the PyTorch `torchscript` (`pt`) format is not used, the `rect` configuration is set to `False`.
- Altered data loader behavior in `train.py` and `val.py` scripts to incorporate the updated `rect` key functionality depending on whether the mode is 'train' or 'val'.

### 🎯 Purpose & Impact
- 🎯 The update provides more configuration options and clarity to users on when and how validation/testing is performed.
- 💡 By enabling the rectangular evaluation by default, validation performance could potentially be improved due to the more efficient use of space on the GPU, leading to faster evaluation times.
- 🚀 Users with only CPU setups or not using `torchscript` will experience default settings more suitable for their configuration, potentially increasing ease of use and performance.

### CODE REVIEW

File: docs/cfg.md [hunk 1]
```diff
@@ -148,7 +149,6 @@ validation dataset and to detect and prevent overfitting.
 
 | Key         | Value | Description                                                                 |
 |-------------|-------|-----------------------------------------------------------------------------|
-| val         | True  | validate/test during training                                               |
```
Comment (Laughing-q) [inline docs/cfg.md:151]: @AyushExel BTW I moved this line to training section as it belongs a training arg.

Comment (AyushExel) [reply docs/cfg.md:151]: okay sounds good

**Comment by AyushExel** [comment]
@Laughing-q looks good to me

**Comment by glenn-jocher** [comment]
@Laughing-q good idea! Merged into updates for 8.0.30 later today or tomorrow.
